MIGRATION.md

```diff
@@ -230,9 +233,27 @@ This was done to remove a hard dependency on `https-proxy-agent` when running
    authorization: 'authorization'
  }`
 
-### 8. Bug fix: Setting an explicit `algorithm` when generating a wallet works now
+### 8. `Wallet` functions default to `ed25519` instead of `secp256k1` signing algorithm
+
+With 3.0, we updated the default signing algorithm used by the Wallet object from secp256k1 to ed25519 in order to default to the higher-performance algorithm. This is a breaking change to all functions used to generate a Wallet, so if you have a pre-existing XRPL account that you're using to generate a specific Wallet using older versions of xrpl.js, you must specify that you are using secp256k1 as the algorithm to decode your private key / seed / etc to get the same behavior as before. See below for specifically how to update your code.
+
+If you are creating new accounts each time (ex. via `Client.fundWallet`), you do not need to specify the signing algorithm.
+
+**Before**
+
+`Wallet.fromSeed('s...')`
+`Wallet.generate()`
+`Wallet.fromEntropy(entropy)`
+`deriveKeyPair(seed="s...")`
+`generateSeed()`
```

Comment:
`generateSeed` don't need to change.

Reply:
if no explicit `algorithm` parameter is provided, `generateSeed` produces a seed of type `secp256k1`. Here's an example unit test: https://github.com/XRPLF/xrpl.js/blob/be732a4a6b7bdf58d0efff084b76a66f3f625fe6/packages/ripple-keypairs/test/api.test.ts#L26

I feel this behavior contradicts the spirit of the update: The default behavior of `generateSeed` function is to produce a `secp256k1` seed, not a `ed25519` seed.

What do you suggest? How can we communicate this?

Reply:
this change could fix this issue: https://github.com/XRPLF/xrpl.js/pull/2658/files#diff-37199a4e4bdd9fb657c619a808e98350642688e38494ad56c36372b43c6437b8L19

but it needs some more time and review

Reply:
That's fair, but you're still usually not providing entropy.

Reply:
okay, fixed in [a16773e](https://github.com/XRPLF/xrpl.js/pull/2659/commits/a16773ee21847ed903fca9f0c6b77098890201fb)